surch/surch.py

```diff
@@ -1,165 +1,335 @@
+########
+# Copyright (c) 2016 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
 import os
 import time
+import logging
+import tempfile
 import subprocess
 
 import yaml
+import click
+import retrying
 import requests
 
-GET_GIT_REPO_DETAILS_API_URL = 'https://api.github.com/orgs/{0}/repos?per_page={1}'
-GIT_PULL_COMMAND = 'git -C {0} pull'
-GIT_CLONE_COMMAND = 'git clone --quiet {0} {1}'
+from . import logger
+
+# This strings is for getting the repo git url list , it get organization, repos_per_page vars
+API_URL_REPO_DETAILS = 'https://api.github.com/orgs/{0}/repos?type={1}&per_page={2}&page={3}'
+# This string is a template for blob_url to redirect you for the problematic commit.
+# It get organization, repo_name, sha, file_name
 BLOB_URL_TEMPLATE = 'https://github.com/{0}/{1}/blob/{2}/{3}'
-GIT_GREP_COMMAND = 'git -C {0} grep -l -e {1} {2}'
-GET_COMMITS_COMMAND = 'git -C {0} rev-list --all'
+# Get organization details
+API_URL_ORGANIZATION_DETAILS = 'https://api.github.com/orgs/{0}'
+
+
+lgr = logger.init()
+
+
+class Surch(object):
+    default_path = os.path.join(tempfile.gettempdir(), 'Surch')
```

Comment:
change to ~/surch/org_name

global var in class capitalized 

and space before init